server/text_generation_server/models/model.py

```diff
@@ -85,6 +85,21 @@ def decode_token(
             return new_text, read_offset, len(all_input_ids)
         else:
             return "", prefix_offset, read_offset
+    
+    def decode_generated_tokens(
+        self,
+        all_input_ids: List[int],
+        num_tokens: int = 0,
+    ) -> str:
+        # Like in `decode_token()`, the prefix text is necessary only to defeat cleanup algorithms in the decode.
+        prefix_text = self.tokenizer.decode(
+            all_input_ids[-num_tokens-1:-num_tokens], skip_special_tokens=False
```

Comment:
I found that the parameters here are different from the original. Can you confirm if there is a need to maintain consistency with the original?

` skip_special_tokens=True, clean_up_tokenization_spaces=False
` 

https://github.com/huggingface/text-generation-inference/blob/main/server/text_generation_server/models/seq2seq_lm.py#L580

Reply:
Thanks for pointing this out!

The decoding parameters in the current main branch are different for `generate()` and `generate_stream()`. Essentially,

- the `generate()` function decodes with `skip_special_tokens=True, clean_up_tokenization_spaces=False`: [https://github.com/huggingface/text-generation-inference/blob/c8a01d759173483efc2135c4e7506b23e14e7fc4/server/text_generation_server/models/flash_causal_lm.py#L796-L799](url);

 - the `generate_stream()` function decodes with `skip_special_tokens=False`:
[https://github.com/huggingface/text-generation-inference/blob/c8a01d759173483efc2135c4e7506b23e14e7fc4/server/text_generation_server/models/model.py#L72-L77](url).

So whether to align our modification with previous `generate()` or `generate_stream()` is well worth discussing.